rust/connlib/tunnel/src/lib.rs

```diff
@@ -296,14 +317,41 @@ impl GatewayTunnel {
 
                     continue;
                 }
-                Poll::Ready(io::Input::UdpDnsQuery(query)) => self.io.send_udp_dns_response(
-                    query.source,
-                    dns::servfail(query.message.for_slice_ref()),
-                )?,
-                Poll::Ready(io::Input::TcpDnsQuery(query)) => self.io.send_tcp_dns_response(
-                    query.source,
-                    dns::servfail(query.message.for_slice_ref()),
-                )?,
+                Poll::Ready(io::Input::UdpDnsQuery(query)) => {
+                    let Some(nameserver) = self.nameservers.first() else {
+                        tracing::info!("No nameserver available to resolve DNS query");
+
+                        self.io.send_udp_dns_response(
+                            query.source,
+                            dns::servfail(query.message.for_slice_ref()),
+                        )?;
+                        continue;
+                    };
+
+                    self.io.send_dns_query(dns::RecursiveQuery::via_udp(
+                        query.source,
+                        SocketAddr::new(*nameserver, dns::DNS_PORT),
+                        query.message.for_slice_ref(),
+                    ));
+                }
+                Poll::Ready(io::Input::TcpDnsQuery(query)) => {
+                    let Some(nameserver) = self.nameservers.first() else {
```

Comment:
I know admins "should have" all nameservers reachable in `/etc/resolv.conf`, but someone is going to have a bad day when this becomes an issue, since otherwise the box will appear to work fine. Firezone will be the only process that can't failover to the other nameservers.

I think we can do a naive approach here and simply forward the query to all of the nameservers list in /etc/resolv.conf, then just send back to the client the first one to respond?

Happy to take a stab at it if you're a little burnt out on the SRV stuff. But I'm fairly confident this is going to bite us (or me... handling support / a lost sale) later on.

Reply:
Did you see my comment above? Can we ship this now like this and make it more resilient later?

Reply:
https://github.com/firezone/firezone/pull/8350#issuecomment-2697580097

Reply:
We can start with a warning that is emitted if we have more than a single DNS server?

Reply:
> Happy to take a stab at it if you're a little burnt out on the SRV stuff. But I'm fairly confident this is going to bite us (or me... handling support / a lost sale) later on.

I am less worried about the implementation complexity but rather the maintenance cost further down the line.

Reply:
What if we test the first nameserver during startup and only pass a single one down? That should be safe and allows us avoid this complexity for now.

Reply:
I had an idea, we just pick the fastets one out of all of them during startup and then use that.